### PR TITLE
Code documentation javascript common.js

### LINKED ISSUE #32: Document wp-admin/js/common.js

### PR DESCRIPTION
**Pull request by rensw90**
Wordpress Trac: [#38309](https://core.trac.wordpress.org/ticket/38309#no0)
Fixes #32

Worked on this together with @benvaassen

### CODE REVIEW

File: src/wp-admin/js/common.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
 /* global setUserSetting, ajaxurl, commonL10n, alert, confirm, pagenow */
 var showNotice, adminMenu, columns, validateForm, screenMeta;
+/**
+ *  @summary
```
Comment (andizer) [inline src/wp-admin/js/common.js]: Missing the summary

File: src/wp-admin/js/common.js [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+ */
 ( function( $, window, undefined ) {
 	var $document = $( document ),
 		$window = $( window ),
 		$body = $( document.body );
 
-// Removed in 3.3.
-// (perhaps) needed for back-compat
+/**
+ * @summary Removed in 3.3.0, needed for back-compatiblity
```
Comment (andizer) [inline src/wp-admin/js/common.js]: Please add a full stop.

File: src/wp-admin/js/common.js [hunk 3]
```diff
@@ -75,19 +143,43 @@ columns = {
...
+ */
 validateForm = function( form ) {
-	return !$( form )
-		.find( '.form-required' )
-		.filter( function() { return $( ':input:visible', this ).val() === ''; } )
-		.addClass( 'form-invalid' )
-		.find( ':input:visible' )
-		.change( function() { $( this ).closest( '.form-invalid' ).removeClass( 'form-invalid' ); } )
-		.length;
+return !$( form )
```
Comment (andizer) [inline src/wp-admin/js/common.js]: Why is the indentation removed.

File: src/wp-admin/js/common.js [hunk 4]
```diff
@@ -97,16 +189,41 @@ showNotice = {
...
+ *
+ * @return {void}
+ */
 screenMeta = {
 	element: null, // #screen-meta
 	toggles: null, // .screen-meta-toggle
 	page:    null, // #wpcontent
 
+	/**
+	 * @summary Initialises the screen meta options panel.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: We are using American English. Please change initialises to initializes.

File: src/wp-admin/js/common.js [hunk 5]
```diff
@@ -140,7 +281,24 @@ screenMeta = {
 		$document.trigger( 'screen:options:open' );
 	},
 
+	/**
+	 * @summary Closes the screen meta options panel.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param panel The screen meta options panel div.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: Missing the type

File: src/wp-admin/js/common.js [hunk 6]
```diff
@@ -140,7 +281,24 @@ screenMeta = {
...
+	 * @param button The toggle button.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: Also missing the type

File: src/wp-admin/js/common.js [hunk 7]
```diff
@@ -298,7 +475,11 @@ $document.ready( function() {
 		// iOS Safari works with touchstart, the rest work with click
 		mobileEvent = isIOS ? 'touchstart' : 'click';
 
-		// close any open submenus when touch/click is not on the menu
+		/**
+		 * @summary Close any open submenus when touch/click is not on the menu.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: I prefer closes instead of close

File: src/wp-admin/js/common.js [hunk 8]
```diff
@@ -260,11 +435,13 @@ $document.ready( function() {
 	$document.on( 'wp-menu-state-set wp-collapse-menu wp-responsive-activate wp-responsive-deactivate', currentMenuItemHasPopup );
 
 	/**
-	 * Ensure an admin submenu is within the visual viewport.
+	 * @summary Ensure an admin submenu is within the visual viewport.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: Ensures...

File: src/wp-admin/js/common.js [hunk 9]
```diff
@@ -244,7 +413,13 @@ $document.ready( function() {
 		$document.trigger( 'wp-collapse-menu', { state: menuState } );
 	});
 
-	// Handle the `aria-haspopup` attribute on the current menu item when it has a sub-menu.
+	/**
+	 * @summary Handle the `aria-haspopup` attribute on the current menu item when it has a sub-menu.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: handles

Comment (andizer) [inline src/wp-admin/js/common.js]: sub-menu should be submenu

File: src/wp-admin/js/common.js [hunk 10]
```diff
@@ -361,19 +559,34 @@ $document.ready( function() {
...
 		$adminmenu.on( 'focus.adminmenu', '.wp-submenu a', function( event ) {
 			if ( $adminmenu.data( 'wp-responsive' ) ) {
 				// The menu is in responsive mode, bail
 				return;
 			}
 
 			$( event.target ).closest( 'li.menu-top' ).addClass( 'opensub' );
+
+			/**
+			 * @summary Closes the submenu on blur from the menu item.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: missing the `@param {Event} event` and `@returns void`

File: src/wp-admin/js/common.js [hunk 11]
```diff
@@ -547,48 +809,98 @@ $document.ready( function() {
...
+	 */
 	$('.search-box input[type="search"], .search-box input[type="submit"]').mousedown(function () {
 		$('select[name^="action"]').val('-1');
 	});
 
-	// Scroll into view when focused
+	/**
+	 * @summary Scroll into view when focus.scroll-into-view is triggered.
+	 *
+	 * @return {void}
```
Comment (andizer) [inline src/wp-admin/js/common.js]: missing docs for `e`

File: src/wp-admin/js/common.js [hunk 12]
```diff
@@ -547,48 +809,98 @@ $document.ready( function() {
...
+			 * action = bulk action dropdown at the top of the table
+			 * action2 = bulk action dropdow at the bottom of the table
+			 */
 			if ( $('select[name="action"]').val() == -1 && $('select[name="action2"]').val() == -1 && pageInput.val() == currentPage )
 				pageInput.val('1');
 		});
 	}
 
+	/**
+	 * @summary Resets the bulk actions when the search button is clicked.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: missing `@returns`

File: src/wp-admin/js/common.js [hunk 13]
```diff
@@ -636,10 +957,12 @@ $document.ready( function() {
 			}
 
 			if ( windowPos > lastScrollPosition ) {
-				// Scrolling down
+				// When a down scroll has been deteceted.
```
Comment (andizer) [inline src/wp-admin/js/common.js]: typo

File: src/wp-admin/js/common.js [hunk 14]
```diff
@@ -751,7 +1100,24 @@ $document.ready( function() {
...
 	}
 
+	/**
+	 * @summary Changes properties of metaboxes and body.
+	 *
+	 * Changes the sortables and responsiveness of metaboxes.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @return {void}
```
Comment (andizer) [inline src/wp-admin/js/common.js]: This is an object which can't return anything.

**Comment by manuelaugustin** [comment]
CR done 👍

**Comment by atimmer** [comment]
Has been done in https://core.trac.wordpress.org/changeset/42351.

**Pull request closed without merging**